File: src/test/java/personal/alexgarland/helloworldmvc/controller/ErrorTestControllerTest.java
package personal.alexgarland.helloworldmvc.controller;

import static org.junit.Assert.*;

import org.junit.Test;
import org.springframework.web.servlet.ModelAndView;

public class ErrorTestControllerTest {

	private ErrorTestController testController = new ErrorTestController();
	
	@Test
	public void getsErrorTestPage() {
		ModelAndView result = testController.getErrorTestPage();
		assertEquals("errorTest", result.getViewName());
		assertEquals(0, result.getModel().size());
	}
	
	@Test(expected = RuntimeException.class)
	public void throwsServerError() {
		testController.generateServerError();
	}

}

File: src/test/kotlin/personal/alexgarland/helloworldmvc/controller/ErrorTestControllerTest.kt
package personal.alexgarland.helloworldmvc.controller

import org.junit.Assert.assertEquals
import org.junit.Test
import org.springframework.web.servlet.ModelAndView

class ErrorTestControllerTest {

    private val testController = ErrorTestController()

    @Test
    fun getsErrorTestPage() {
        val result = testController.errorTestPage
        assertEquals("errorTest", result.viewName)
        assertEquals(0, result.model.size.toLong())
    }

    @Test(expected = RuntimeException::class)
    fun throwsServerError() {
        testController.generateServerError()
    }

}
